webpack/assets/javascripts/react_app/components/HostDetails/DetailsCard/PowerStatusDropdown.js

```diff
@@ -0,0 +1,73 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { Dropdown, DropdownToggle, DropdownItem } from '@patternfly/react-core';
+
+import { capitalize, foremanUrl } from '../../../common/helpers';
+import { useAPI } from '../../../common/hooks/API/APIHooks';
+import PowerStatusIcon from './PowerStatusIcon';
+import { POWER_REQUEST_OPTIONS, SUPPORTED_POWER_STATES } from './constant';
+import { changeHostPower } from './actions';
+import './styles.scss';
+
+const PowerStatusDropDown = ({ hostID, hasPowerPermission }) => {
+  const powerURL = hostID && foremanUrl(`/api/hosts/${hostID}/power`);
+  const {
+    response: { state: currentState, statusText, title },
+    status: responseStatus,
+  } = useAPI('get', powerURL, POWER_REQUEST_OPTIONS);
+
+  const dispatch = useDispatch();
+  const [isOpen, setOpen] = useState(false);
+  const changePowerHandler = targetState => {
+    dispatch(changeHostPower(targetState, hostID));
+  };
+  const dropdownItems = () =>
+    SUPPORTED_POWER_STATES.map(state => (
+      <DropdownItem
+        onClick={() => changePowerHandler(state)}
+        isDisabled={state === currentState}
+        key={state}
+      >
+        {capitalize(state)}
```

Comment:
Would this work with translations? Why not capitalize the original string instead of doing it at runtime?

Reply:
> Why not capitalize the original string instead of doing it at runtime
we can, but the server needs a lowercase `state` I think it easier to do `toLowerCase`for servers params and translate the original string 👍

Reply:
I would actually make `SUPPORTED_POWER_STATES` a mapping/list of tuples: the key is the constant that the server uses, the value is the label that should also be translated.